src/isl/isl_type.rs

```diff
@@ -1,115 +1,50 @@
-                None => {
-                    return Err(invalid_schema_error_raw(
-                        "A type name symbol token does not have any text",
-                    ))
-                }
-            };
-            // TODO: add more constraints to match below
-            let constraint = match constraint_name {
-                "all_of" => {
-                    //TODO: create a method/macro for this ion type check which can be reused
-                    if value.ion_type() != IonType::List {
-                        return Err(invalid_schema_error_raw(format!(
-                            "all_of constraint was a {:?} instead of a list",
-                            value.ion_type()
-                        )));
-                    }
-                    let types: Vec<IslTypeRef> = value
-                        .as_sequence()
-                        .unwrap()
-                        .iter()
-                        .map(|e| IslTypeRef::parse_from_ion_element(e))
-                        .collect::<IonSchemaResult<Vec<IslTypeRef>>>()?;
-                    IslConstraint::AllOf(types)
-                }
-                "type" => {
-                    if value.ion_type() != IonType::Symbol && value.ion_type() != IonType::Struct {
-                        return Err(invalid_schema_error_raw(format!(
-                            "type constraint was a {:?} instead of a symbol/struct",
-                            value.ion_type()
-                        )));
-                    }
-                    let type_reference: IslTypeRef = IslTypeRef::parse_from_ion_element(value)?;
-                    IslConstraint::Type(type_reference)
-                }
-                _ => {
-                    return Err(invalid_schema_error_raw(format!(
-                        "Type: {:?}
-                            can not be built as constraint: {}
-                            does not exist",
-                        ion_struct, constraint_name
-                    )))
-                }
-            };
-            constraints.push(constraint);
+    /// Provides the underlying constraints of [IslTypeImpl]
+    pub fn constraints(&self) -> &[IslConstraint] {
+        match &self {
+            IslType::Named(named_type) => named_type.constraints(),
+            IslType::Anonymous(anonymous_type) => anonymous_type.constraints(),
         }
-        Ok(AnonymousIslType::new(constraints))
-    }
-}
-
-/// Provides an implementation of PartialEq to compare two IslTypes
-impl PartialEq for AnonymousIslType {
-    fn eq(&self, other: &Self) -> bool {
-        self.constraints.eq(&other.constraints)
     }
 }
 
-/// Represents a public facing schema type [NamedIslType] which can be converted to a solid [NamedTypeDefinition] using TypeStore
+/// Represents both named and anonymous [IslType] which can be converted to a solid [TypeDefinition] using TypeStore
 /// named ISL type grammar: `type:: { name: <NAME>, <CONSTRAINT>...}`
```

Comment:
```suggestion
/// Represents both named and anonymous [IslType]s and can be converted to a solid [TypeDefinition] using [TypeStore].
/// Named ISL type grammar: `type:: { name: <NAME>, <CONSTRAINT>...}`
```
As I was reading this my brain turned the two lines into one big sentence: "...using TypeStore named ISL type grammar: `type::`..." 😛